### ISSUE TITLE
Rebuild with my modified quic-go 

### ISSUE THREAD

**Issue by ckkk1024** [caddyserver/caddy#5658]
I am a novice in caddy, and I want to use the quic/http3 function in caddy, so I compile and install caddy from the source code according to the README document of the caddy repository. I understand that during this process, caddy will pull the quic-go source code locally as a dependency, and then compile it together. Is my understanding correct?
And I want to modify the quic-go source code pulled above, such as customizing some log functions, and then make it take effect in caddy, how should I do it? 

Thanks！

**Comment by mholt** [comment]
Yes, in general Go works this way: dependencies are downloaded to a read-only local cache, and build from that.

If you need to modify a dependency, the usual pattern is to add a `replace` to the go.mod file to point to your locally-cloned copy of it (from, say, `git clone`):

```
replace <module> => ./path-to-module
```

See the Go docs: https://go.dev/ref/mod#go-mod-file-replace
